packages/react/src/auto/shadcn/inputs/ShadcnAutoEncryptedStringInput.tsx

```diff
@@ -0,0 +1,32 @@
+import { EyeIcon, EyeOffIcon } from "lucide-react";
+import React, { useState } from "react";
+import type { Control } from "../../../useActionForm.js";
+import { autoInput } from "../../AutoInput.js";
+import type { ShadcnElements } from "../elements.js";
+import { makeShadcnAutoTextInput } from "./ShadcnAutoTextInput.js";
+
+export const makeShadcnAutoEncryptedStringInput = ({ Input, Label, Button }: Pick<ShadcnElements, "Input" | "Label" | "Button">) => {
+  const TextInput = makeShadcnAutoTextInput({ Input, Label });
+
+  function ShadcnAutoEncryptedStringInput(props: { field: string; control?: Control<any>; className?: string; suffix?: React.ReactNode }) {
+    const [isShown, setIsShown] = useState(false);
+    const { suffix, ...restProps } = props;
+
+    const showHideToggleButton = (
```

Comment:
@airhorns should users be supplying their password input mask / unmask buttons ? or this okay

Reply:
No, its ok that they can't. As long as you use the Button element I am ok with this kind of thing. Its a kinda arbitrary decision we've made on a case by case basis, but this is fine for now and we can accept future PRs adding more customization.